### ISSUE TITLE
Support TI COFF1 and TI COFF2 formats

### ISSUE THREAD

**Issue by XVilka** [rizinorg/rizin#4299]
Extend current COFF implementation support also:
- [ ] Texas Instruments COFF1 format
- [ ] Texas Instruments COFF2 format

See this documentation:
- https://www.ti.com/lit/an/spraao8/spraao8.pdf?ts=1708995439636
- https://www.ti.com/lit/an/spra573/spra573.pdf?ts=1708995446847
- https://www.ti.com/sc/docs/products/micro/msp430/userguid/as_2.pdf
- https://gist.github.com/1073231/aa57bf1599fe4f3bd417492cd5bde097e7f31139

**Comment by notxvilka** [comment]
@historicattle yes, it's possible. @wargio followed the same approach with the ECOFF: https://github.com/rizinorg/rizin/pull/5447

If we determine common code between these formats, we can extract and deduplicate in the future.

**Comment by historicattle** [comment]
Okay thanks

**Comment by olowoyobabajide** [comment]
I want to work on this task
